babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -505,13 +507,56 @@ SqlCreate SqlCreateTable(Span s, boolean replace) :
     |
         { query = null; }
     )
+    (
+       index = SqlCreateTableIndex(s)
+       {
+           if(index instanceof SqlPrimaryIndex){
+               primaryIndex = (SqlPrimaryIndex) index;
+           }
+           //code for non-primary index support can be added here
+       }
+    )*
     onCommitType = OnCommitTypeOpt()
     {
         return new SqlCreateTable(s.end(this), replace, setType, volatility, ifNotExists, id,
-            tableAttributes, columnList, query, withData, onCommitType);
+            tableAttributes, columnList, query, withData, primaryIndex, onCommitType);
     }
 }
 
+/**
+Parses an index declaration.
+Currently only supports PRIMARY INDEX statements, but can be extended to support non-primary indices
+*/
+SqlIndex SqlCreateTableIndex(Span s) :
+{
+   List<SqlNode> columns;
+   SqlIdentifier name = null;
+   boolean isUnique = false;
+}
+{
+   (
+       <NO> <PRIMARY> <INDEX>
+       {
+           return new SqlPrimaryIndex(s.end(this), /*columns*/ null, /*name*/ null, /*isUnique*/ false, /*explicitNoPrimaryIndex*/ true);
```

Comment:
nit: for the comments specifying null being passed in use "=" as well. So instead of /\*columns\*/ do /\*columns=\*/